tests/src/nodes/mixnode.rs

```diff
@@ -14,21 +16,32 @@ use crate::{get_available_port, MixnetConfig};
 const MIXNODE_BIN: &str = "../target/debug/mixnode";
 
 pub struct MixNode {
+    _tempdir: tempfile::TempDir,
     child: Child,
 }
 
 impl Drop for MixNode {
     fn drop(&mut self) {
+        if std::thread::panicking() {
+            persist_tempdir(&mut self._tempdir, "mixnode");
```

Comment:
I'm going to add a more specific label here, to distinguish between multiple nodes.

Reply:
I'm not so sure having all this stuff while dropping. I would have to check, but I think having code that can panic when droping is not a good idea.

Reply:
I agree. I haven't noticed it. How about just printing an error and ignore it if the error occurs since this is only for testing? We have two panic points in the drop function:
```rust
let dir = std::mem::replace(tempdir, tempfile::tempdir().unwrap());
``` 
and 
```rust
self.child.kill().unwrap();
```
The first one will rarely fail. I guess the second also doesn't fail in most cases since it just send SIGKILL to the process.

I haven't tried to spawn a task or a thread in the drop function, but even though we use a task/thread, I think there is no other way than ignore errors inside the task.

Reply:
@danielSanchezQ I just made it not panic in the drop function. Instead, I made it print an error log and ignore it. That would be enough for testing as I mentioned above. What do you think about this?

[ecfbe6e](https://github.com/logos-co/nomos-node/pull/510/commits/ecfbe6e1e53c51ab92e78a4f720e80ada4cf4e65)

Reply:
@danielSanchezQ Could you have a quick look for this change when you have chance?